geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisKeyJUnitTest.java

```diff
@@ -39,43 +40,45 @@ public static void classSetup() {
   @Test
   public void testRoutingId_withHashtags() {
     RedisKey key = new RedisKey(stringToBytes("name{user1000}"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("user1000"));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("user1000") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("{user1000"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("{user1000"));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("{user1000") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("}user1000{"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("}user1000{"));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("}user1000{") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("user{}1000"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("user{}1000"));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("user{}1000") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("user}{1000"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("user}{1000"));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("user}{1000") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("{user1000}}bar"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("user1000"));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("user1000") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("foo{user1000}{bar}"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("user1000"));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("user1000") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("foo{}{user1000}"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("foo{}{user1000}"));
+    assertThat(key.getSlot())
+        .isEqualTo(CRC16.calculate("foo{}{user1000}") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("{}{user1000}"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("{}{user1000}"));
+    assertThat(key.getSlot())
+        .isEqualTo(CRC16.calculate("{}{user1000}") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(stringToBytes("foo{{user1000}}bar"));
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate("{user1000"));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("{user1000") % RegionProvider.REDIS_SLOTS);
 
     key = new RedisKey(new byte[] {});
-    assertThat(key.getCrc16()).isEqualTo(CRC16.calculate(""));
+    assertThat(key.getSlot()).isEqualTo(CRC16.calculate("") % RegionProvider.REDIS_SLOTS);
   }
 
   @Test
   public void testSerialization_withPositiveSignedShortCRC16() throws Exception {
     RedisKey keyOut = new RedisKey(stringToBytes("012345"));
-    assertThat((short) keyOut.getCrc16()).isPositive();
+    assertThat((short) keyOut.getSlot()).isPositive();
```

Comment:
should this just be "testSerialization" and not worry about the slot being positive as a short? If we wanted to test that slot is always >= 0 it seems like it should be in its own test method.

Reply:
done